### PR TITLE
Sled tc dispatcher

### PR DESCRIPTION
**Pull request by astoycos**
Finishes https://github.com/bpfman/bpfman/issues/860 depends on the changes in #901  (I'll leave as a draft until that work merges)

### CODE REVIEW

File: bpfman-api/src/config.rs [hunk 1]
```diff
@@ -63,6 +63,15 @@ impl XdpMode {
             XdpMode::Hw => XdpFlags::HW_MODE,
         }
     }
+
+    pub fn from_u32(val: u32) -> Self {
```
Comment (astoycos) [inline bpfman-api/src/config.rs:67]: Convert to tryfrom

Comment (dave-tucker) [reply bpfman-api/src/config.rs:69]: Thanks! Saved me adding that comment here lol

File: bpfman/src/utils.rs [hunk 2]
```diff
@@ -81,6 +81,14 @@ pub(crate) fn bytes_to_u32(bytes: Vec<u8>) -> u32 {
     )
 }
 
+pub(crate) fn bytes_to_u16(bytes: Vec<u8>) -> u16 {
+    u16::from_ne_bytes(
+        bytes
+            .try_into()
+            .expect("unable to martial &[u8] to &[u8; 4]"),
```
Comment (dave-tucker) [inline bpfman/src/utils.rs:88]: Just saw this one, it should be `Marshall` not `Martial`

**Review by dave-tucker — CHANGES_REQUESTED** [review]

**Review by dave-tucker — CHANGES_REQUESTED** [review]
Couple of nits and the `TryFrom` to do, but otherwise LGTM

File: bpfman/src/utils.rs (hunk 2 above)
Comment (astoycos) [reply bpfman/src/utils.rs]: Good catch :)

**Comment by astoycos** [comment]
Done!

**Review by dave-tucker — APPROVED** [review]

**Pull request merged**